tsdb/engine/tsm1/tombstone.go

```diff
@@ -181,30 +186,40 @@ func (t *Tombstoner) writeTombstone(tombstones []Tombstone) error {
 
 	bw := bufio.NewWriterSize(tmp, 1024*1024)
 
-	binary.BigEndian.PutUint32(b[:4], v2header)
+	binary.BigEndian.PutUint32(b[:4], v3header)
 	if _, err := bw.Write(b[:4]); err != nil {
 		return err
 	}
 
+	if err := bw.Flush(); err != nil {
```

Comment:
Yes.  It might not be needed though. I was getting some weird decompresssion results when testing and added this, but it was actually that I was not using `io.ReadFull`